### PR TITLE
Init: remove unsafe characters from  recursively

### PR DESCRIPTION
**Pull request by klees**
This allows $_GET-parameters to be nested more then one level.

### CODE REVIEW

File: Services/Init/classes/class.ilInitialisation.php [hunk 1]
```diff
@@ -37,18 +37,23 @@ protected static function removeUnsafeCharacters()
...
-				$_GET[$k] = str_replace(array("\x00", "\n", "\r", "\\", "'", '"', "\x1a"), "", $v);
+		self::recursivelyRemoveUnsafeCharacters($_GET);
+	}
 
-				// this one is for XSS of any kind
-				$_GET[$k] = strip_tags($_GET[$k]);
+	protected static function recursivelyRemoveUnsafeCharacters(&$var) {
+		if (is_array($var)) {
+			foreach ($var as $k => $v) {
+				$var[$k] = self::recursivelyRemoveUnsafeCharacters($v);
```
Comment (smeyer-ilias) [inline Services/Init/classes/class.ilInitialisation.php]: The key is not purged and may contain unsafe characters.
What do think about purging them, too?

**Review by smeyer-ilias — COMMENTED** [review]
see comment about key handling.

File: Services/Init/classes/class.ilInitialisation.php (hunk 1 above)
Comment (klees) [reply Services/Init/classes/class.ilInitialisation.php]: Done.

Comment (klees) [reply Services/Init/classes/class.ilInitialisation.php]: Done.

Comment (smeyer-ilias) [reply Services/Init/classes/class.ilInitialisation.php]: unset($var[$k]);
otherwise the old (unsanitized) keys remain in the array.

**Comment by smeyer-ilias** [comment]
Merged. Thanks

**Pull request merged**